### PR TITLE
Only log the $_SERVER information under YII_DEBUG

### PR DESCRIPTION
**Pull request by CoolGoose**
Since $_SERVER can contain sensitive information that could leak in CI pipelines ideally it should happen under the YII_DEBUG flag only

| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  |  ❌
| Breaks BC?    | Maybe if people actually want the information without having the YII_DEBUG on
| Tests pass?   | ✔️
| Fixed issues  | N/A

### CODE REVIEW

**Comment by samdark** [comment]
Would you please elaborate about "could leak in CI pipelines"?

**Comment by CoolGoose** [comment]
@samdark Sorry for the late reply. Basically if you have an error that triggers the logger, in your CI pipeline (in my case gitlab) you'll see the information from $_SERVER which can contain sensitive information.

**Comment by samdark** [comment]
Ah. Got it. That's valid case.

**Comment by samdark** [comment]
Would you please add a line for CHANGELOG and a note for UPGRADE? Thanks.

**Comment by CoolGoose** [comment]
@samdark I apologize in advance but it seems I am having a slightly brain dead moment, but I can't find a changelog / upgrade file. 😄 

Purely text wise.

CHANGELOG:
* Modified the base error handler to not expose $_SERVER details unless YII_DEBUG is set

UPGRADE:
In order to keep displaying the $_SERVER information in the base error handler make sure your YII_DEBUG flag is set to true.

**Comment by samdark** [comment]
- https://github.com/yiisoft/yii2/blob/master/framework/UPGRADE.md
- https://github.com/yiisoft/yii2/blob/master/framework/CHANGELOG.md

**Comment by CoolGoose** [comment]
And added, Friday nights aren't a good period to search for things.
Let me know if they seem to be correct or not and thank you for the help.

File: framework/UPGRADE.md [hunk 1]
```diff
@@ -58,6 +58,9 @@ Upgrade from Yii 2.0.41
   white spaces anymore. If your application expects non-trimmed values provided to this validator make sure to trim 
   them first (i.e. by using `trim` / `filter` "validators").
 
+* `yii\base\ErrorHandler` does not expose the `$_SERVER` information implicitely anymore. If your application still expects
```
Comment (bizley) [inline framework/UPGRADE.md:61]: It should go under new paragraph `Upgrade from Yii 2.0.42`

**Comment by CoolGoose** [comment]
Do you want the PR squashed or are you going to squash on merge ? 
And again thanks for the patience.

File: framework/UPGRADE.md [hunk 2]
```diff
@@ -51,6 +51,12 @@ if you want to upgrade from version A to version C and there is
 version B between A and C, you need to follow the instructions
 for both A and B.
 
+Upgrade from Yii 2.0.42
+-----------------------
+
+* `yii\base\ErrorHandler` does not expose the `$_SERVER` information implicitely anymore. If your application still expects
+  the presence of the information you can set `YII_DEBUG` to true.
```
Comment (samdark) [inline framework/UPGRADE.md:58]: That's not a good advice in case the application expects it in production mode. I'd just remove this "If".

Comment (samdark) [inline framework/UPGRADE.md:58]:
```suggestion
* `yii\base\ErrorHandler` does not expose the `$_SERVER` information implicitely anymore.
```

**Comment by samdark** [comment]
@CoolGoose squashin on merge is alright.

**Review by samdark — APPROVED** [review]

**Comment by bizley** [comment]
Thank you

**Comment by CoolGoose** [comment]
Thank you as well.

**Pull request merged**